### PR TITLE
Fixed some CS1572, 1573, 1574 and 1591 warnings

### PR DESCRIPTION
**Pull request by d3fkn1ght**
﻿## Description of Change

Fixed some CS1572, 1573, 1574 and 1591 warnings

### Bugs Fixed

- Adding some missing XML Docs - issue #1978

### API Changes

None

### Behavioral Changes

None

### PR Checklist

- [ ] Has tests (if omitted, state reason in description)
- [X] Rebased on top of master at time of PR
- [] Changes adhere to coding standard

### CODE REVIEW

File: src/Prism.Core/Modularity/IModuleInfo.cs [hunk 1]
```diff
@@ -2,13 +2,36 @@
...
+    /// <summary>
+    /// Set of properties for each Module
+    /// </summary>
+public interface IModuleInfo : IModuleCatalogItem
     {
+        /// <summary>
+        /// The modules this instance depends on.
+        /// </summary>
         Collection<string> DependsOn { get; set; }
+        /// <summary>
```
Comment (bartlannoeye) [inline src/Prism.Core/Modularity/IModuleInfo.cs:14]: Newline between previous property and comment

File: src/Prism.Core/Modularity/IModuleInfo.cs [hunk 2]
```diff
@@ -2,13 +2,36 @@
...
+        /// </summary>
         string ModuleType { get; set; }
+        /// <summary>
+        /// A string ref for the module for registration/resolution
+        /// </summary>
         string Ref { get; set; }
+        /// <summary>
+        /// An enum that lists the modules state
+        /// <see cref="ModuleState"/>
+        /// </summary>
```
Comment (dansiegel) [inline src/Prism.Core/Modularity/IModuleInfo.cs:34]: Properties really should follow the Gets or Sets {description} format... in this case

```
/// Gets or Sets the current <see cref="ModuleState" />
```

File: src/Prism.Core/Modularity/IModuleInfoGroup.cs [hunk 3]
```diff
@@ -4,10 +4,20 @@
...
     {
+        /// <summary>
+        /// When Prism should Initialize the module
+        /// <see cref="InitializationMode"/>
+        /// </summary>
         InitializationMode InitializationMode { get; set; }
 
+        /// <summary>
+        /// A string ref for the module for registration/resolution
+        /// </summary>
```
Comment (dansiegel) [inline src/Prism.Core/Modularity/IModuleInfoGroup.cs:20]: See Ref notes from IModuleInfo

File: src/Prism.Core/Common/ParametersBase.cs [hunk 4]
```diff
@@ -7,14 +7,24 @@
 
 namespace Prism.Common
 {
+    /// <summary>
+    /// Base class for Navigation parameters
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs]: I have update the comment with your suggested changes

File: src/Prism.Core/Common/ParametersBase.cs [hunk 5]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
         public int Count => _entries.Count;
 
+        /// <summary>
+        /// Returns an IEnumerable of the Keys in the collection
+        /// </summary>
         public IEnumerable<string> Keys => 
             _entries.Select(x => x.Key);
 
+        /// <summary>
+        /// Adds the KeyValuePair to the Collection
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs]: I have updated the comments to reduce confusion identified by @dansiegel

File: src/Prism.Core/Common/ParametersBase.cs [hunk 6]
```diff
@@ -58,6 +68,12 @@ protected ParametersBase(string query)
             }
         }
 
+        /// <summary>
+        /// Searches Parameter collection and returns value if Collection contains key.
+        /// Otherswise returns null.
+        /// </summary>
+        /// <param name="key"></param>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs]: I have added text to the param and returns tags

File: src/Prism.Core/Common/ParametersBase.cs [hunk 7]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
+        /// Returns an IEnumerable of the Keys in the collection
+        /// </summary>
         public IEnumerable<string> Keys => 
             _entries.Select(x => x.Key);
 
+        /// <summary>
+        /// Adds the KeyValuePair to the Collection
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs]: I have added text for the key and value <param> tags

File: src/Prism.Core/Common/ParametersBase.cs [hunk 8]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
+        /// <returns>Enumerator</returns>
         public IEnumerator<KeyValuePair<string, object>> GetEnumerator() =>
             _entries.GetEnumerator();
-
+        /// <summary>
+        /// Returns the value of the member referenced by key
+        /// </summary>
+        /// <typeparam name="T">The type of object to be returned</typeparam>
+        /// <param name="key"></param>
+        /// <returns></returns>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs]: I have added text to the empty tag

File: src/Prism.Core/Modularity/IModuleInfo.cs [hunk 9]
```diff
@@ -2,13 +2,36 @@
...
 namespace Prism.Modularity
 {
-    public interface IModuleInfo : IModuleCatalogItem
+    /// <summary>
+    /// Set of properties for each Module
+    /// </summary>
+public interface IModuleInfo : IModuleCatalogItem
     {
+        /// <summary>
+        /// The modules this instance depends on.
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Modularity/IModuleInfo.cs]: I have added the requested text of names to the comment
The module names this instance depends on.

File: src/Prism.Core/Modularity/IModuleInfo.cs [hunk 10]
```diff
@@ -2,13 +2,36 @@
...
+        /// </summary>
         Collection<string> DependsOn { get; set; }
+        /// <summary>
+        /// When Prism should Initialize the module
+        /// <see cref="InitializationMode"/>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Modularity/IModuleInfo.cs]: I have update the class with the request comment

File: src/Prism.Core/Modularity/IModuleInfo.cs [hunk 11]
```diff
@@ -2,13 +2,36 @@
 
 namespace Prism.Modularity
 {
-    public interface IModuleInfo : IModuleCatalogItem
+    /// <summary>
+    /// Set of properties for each Module
+    /// </summary>
+public interface IModuleInfo : IModuleCatalogItem
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Modularity/IModuleInfo.cs]: Indentation fixed

File: src/Prism.Core/Modularity/IModuleInfo.cs [hunk 12]
```diff
@@ -2,13 +2,36 @@
...
+        /// The name of the module
+        /// </summary>
         string ModuleName { get; set; }
+        /// <summary>
+        /// The module's type
+        /// </summary>
         string ModuleType { get; set; }
+        /// <summary>
+        /// A string ref for the module for registration/resolution
+        /// </summary>
```
[the remark this answers is not in the document]
Comment (d3fkn1ght) [reply src/Prism.Core/Modularity/IModuleInfo.cs]: I have made the requested changes

File: src/Prism.Core/Modularity/IModuleInfo.cs (hunk 2 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Modularity/IModuleInfo.cs]: I have made the requested changes

**Comment by d3fkn1ght** [comment]
I have made all the requested changes.  Please review at your convenience.

File: src/Prism.Core/Common/ParametersBase.cs [hunk 13]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
+        /// <param name="key">The key to reference this value in the KeyValuePair<string, object></param>
+        /// <param name="value">The value of the parameter to store</param>
         public void Add(string key, object value) =>
             _entries.Add(new KeyValuePair<string, object>(key, value));
 
+        /// <summary>
+        /// Checks collection for presense of key
+        /// </summary>
+        /// <param name="key">The key to check in the collection</param>
+        /// <returns>True if key exists; else returns false.</returns>
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:116]: Please use: `<c>true</c> if key exists; else returns <c>false</c>`

File: src/Prism.Core/Common/ParametersBase.cs [hunk 14]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
+        /// <returns>Enumerator</returns>
         public IEnumerator<KeyValuePair<string, object>> GetEnumerator() =>
             _entries.GetEnumerator();
-
+        /// <summary>
+        /// Returns the value of the member referenced by key
+        /// </summary>
+        /// <typeparam name="T">The type of object to be returned</typeparam>
+        /// <param name="key">The key for the value to be returned</param>
+        /// <returns>Returns a matching parameter of <typeparam name="T" /> if one exists in the Collection</returns>
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:131]: Use `<see cref="T" />` not `<typeparam name="T" />`

File: src/Prism.Core/Common/ParametersBase.cs [hunk 15]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
         public int Count => _entries.Count;
 
+        /// <summary>
+        /// Returns an IEnumerable of the Keys in the collection
+        /// </summary>
         public IEnumerable<string> Keys => 
             _entries.Select(x => x.Key);
 
+        /// <summary>
+        /// Adds the <paramref name="key" /> and <paramref name="value" /> to the KeyValuePair<string,object> collection
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:105]: Use `<see cref="T" />` not `<typeparam name="T" />`

File: src/Prism.Core/Common/ParametersBase.cs [hunk 16]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
+        /// <param name="key">The key for the value to be returned</param>
+        /// <returns>Returns a matching parameter of <typeparam name="T" /> if one exists in the Collection</returns>
         public T GetValue<T>(string key) => 
             _entries.GetValue<T>(key);
 
+        /// <summary>
+        /// Returns an IEnumerable of all parameters 
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="key"></param>
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:139]: these are still empty

File: src/Prism.Core/Common/ParametersBase.cs [hunk 17]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
         public IEnumerable<T> GetValues<T>(string key) =>
             _entries.GetValues<T>(key);
 
+        /// <summary>
+        /// Checks to see if the parameter collection contains the value
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="key"></param>
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:147]: these are still empty

File: src/Prism.Core/Common/ParametersBase.cs [hunk 18]
```diff
@@ -74,32 +90,72 @@ protected ParametersBase(string query)
...
         public bool TryGetValue<T>(string key, out T value) =>
             _entries.TryGetValue(key, out value);
 
         IEnumerator IEnumerable.GetEnumerator() =>
             GetEnumerator();
 
+        /// <summary>
+        /// Converts parameter collection to a parameter string
+        /// </summary>
+        /// <returns></returns>
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:158]: this is still empty

File: src/Prism.Core/Common/ParametersBase.cs [hunk 19]
```diff
@@ -129,6 +185,10 @@ public override string ToString()
             return queryBuilder.ToString();
         }
 
+        /// <summary>
+        /// Adds a collection of parameters to the local parameter list
+        /// </summary>
+        /// <param name="parameters"></param>
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:191]: this is still empty

File: src/Prism.Core/Common/ParametersBase.cs (hunk 17 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs:147]: I have added a text to the tag

File: src/Prism.Core/Common/ParametersBase.cs (hunk 13 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs:116]: I have added the reqeusted changes.  Not sure if you wanted to keep the <returns> tag but I did for now.

File: src/Prism.Core/Common/ParametersBase.cs (hunk 14 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs:131]: I have incorporated the requested change

File: src/Prism.Core/Common/ParametersBase.cs (hunk 16 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs:139]: I have added text to the tag

File: src/Prism.Core/Common/ParametersBase.cs (hunk 18 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs:158]: I have added text to the tag

File: src/Prism.Core/Common/ParametersBase.cs (hunk 19 above)
Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs:191]: I have added text to the tag

**Comment by d3fkn1ght** [comment]
I believe I have made all the required changes

File: src/Prism.Core/Common/ParametersBase.cs [hunk 20]
```diff
@@ -58,6 +68,12 @@ protected ParametersBase(string query)
             }
         }
 
+        /// <summary>
+        /// Searches Parameter collection and returns value if Collection contains key.
+        /// Otherswise returns null.
+        /// </summary>
+        /// <param name="key">The key for the value to be returned</param>
+        /// <returns></returns>
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:76]: this is still empty... be sure all elements have a description

Comment (d3fkn1ght) [reply src/Prism.Core/Common/ParametersBase.cs]: I beleive i have finally got all the tags filled ... sorry

**Comment by d3fkn1ght** [comment]
Hopefull all the tags have text now.  I apologize for missing that.

File: src/Prism.Core/Common/ParametersBase.cs [hunk 21]
```diff
@@ -74,32 +90,74 @@ protected ParametersBase(string query)
...
         public int Count => _entries.Count;
 
+        /// <summary>
+        /// Returns an IEnumerable of the Keys in the collection
+        /// </summary>
         public IEnumerable<string> Keys => 
             _entries.Select(x => x.Key);
 
+        /// <summary>
+        /// Adds the <paramref name="key" /> and <paramref name="value" /> to the KeyValuePair<string,object> collection
```
Comment (dansiegel) [inline src/Prism.Core/Common/ParametersBase.cs:105]: let's remove the paramref's from here...

**Pull request merged**